modules/era/lua/era_HNM_system.lua

```diff
@@ -15,249 +14,250 @@ local behemothDomID   = require("scripts/zones/Behemoths_Dominion/IDs")
 -----------------------------------
 -- Module definition
 -----------------------------------
+-- This module attempts to replicate old Land King pop system in an era-accurate way.
+-- It comes along a ToD perpetuation/retainment system, for server crashes, since this NMs were usually contested by endgame LSs.
 local hnmSystem = Module:new("era_HNM_System")
 
--- NOTE: This module attempts to replicate old Land King pop system in an era-accurate way.
--- It comes along a ToD perpetuation/retainment system, for server crashes, since this NMs were usually contested by endgame LSs.
+-----------------------------------
+-- Module enable/disable
+-----------------------------------
 -- Do not use along custom_HNM_System module. Choose one or the other.
+hnmSystem:setEnabled(false)
 
 ----------------------------------------------------------------------------------------------------
 -- Dragon's Aery: Fafnir, Nidhogg
 ----------------------------------------------------------------------------------------------------
-
 hnmSystem:addOverride("xi.zones.Dragons_Aery.Zone.onInitialize", function(zone)
+    super(zone)
+
     local hnmPopTime   = GetServerVariable("[HNM]Fafnir")   -- Time the NM will spawn at.
     local hnmKillCount = GetServerVariable("[HNM]Fafnir_C") -- Number of times NQ King has been slain in a row.
+    local currentTime  = os.time()
 
     -- First-time setup.
     if hnmPopTime == 0 then
-        hnmPopTime = os.time() + math.random(1, 48) * 1800
+        hnmPopTime = currentTime + math.random(1, 48) * 1800
```

Comment:
For the era module (maybe the custom too, idk) I believe the window is only supposed to be 3 hours instead of 24 from when servers come up, cutting the random roll down to 6 from 48.

Reply:
Correct.

This section is the "first time" setup. It's there in case there is no server variable, either becouse the system has never been used or becouse it has been deleted on server operator error. This makes it so they dont immediately pop in said cases.

The onMobDespawn sections of each nm contain the regular windows

```

    local randomPopTime = 75600 + math.random(0, 6) * 1800 -- 21 hours minimum pop time. 7 windows. 1 every 30 mins.

    SetServerVariable("[HNM]Fafnir", os.time() + randomPopTime) -- Save next pop time.
    
    GetMobByID(dragonsAeryID.mob.FAFNIR):setRespawnTime(randomPopTime) -- Sets the respawn countdown
```

Reply:
I was thinking server ops could delete the vars during scheduled downtime to give the old retail experience of logging out in aery before downtime to catch the early windows, while also catching unexpected crashes and persisting the ToD. The catching unexpected crashes looks like it works as intended but there isn't really a way to simulate that old maintenance camping with this setup. It would still result in a minimum 30 min downtime but only a max 3 hours instead of 24 hours of possible windows.

Reply:
Ultimately, this is an optional module and server ops are gonna edit as needed but preventing abuse of crashed was a highly desired feature that lead to lots of help requests from people wanting to know how to do it (which means lead to lots of answering the same questions).

Myself on my own server got a step the other direction and instead of trying to retain the prior time remaining from before a crash, I punish the entire world by resetting everything to its full cooldown so that server crashing makes _nobody_ a winner (seen to many people actively seek new ways to crash a server just to make themselves get the claim over someone else, even when the ToD wasn't an issue - and that made me a leeeetle angry).